libraries/AP_GPS/AP_GPS_NMEA.cpp

```diff
@@ -116,6 +118,8 @@ bool AP_GPS_NMEA::_decode(char c)
     if (!_is_checksum_term)
         _parity ^= c;
 
+    _sentence_length++;
```

Comment:
Why not just do this at the start of `_decode()` (The one in the first half the switch can then be removed)